src/Control/Monad/ST/Internal.purs

```diff
@@ -8,7 +8,7 @@ import Partial.Unsafe (unsafePartial)
 -- | `ST` is concerned with _restricted_ mutation. Mutation is restricted to a
 -- | _region_ of mutable references. This kind is inhabited by phantom types
 -- | which represent regions in the type system.
-foreign import kind Region
+data Region
```

Comment:
I think maybe we want to define this as `foreign import data Region` (@natefaubion?)

Reply:
This is fine because it's purely type-level. I actually brought up in the PR that `foreign import kind` should probably just be empty data since it's meant to be uninhabited.